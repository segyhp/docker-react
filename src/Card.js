import React from 'react';

// const Card = (props) => {
    const Card = ({name, email, id}) => {

    //using destructuring
    // const  {name, email, id} = props;
    return (
        <div className='tc bg-light-green dib br3 pd3 ma2 grow bw2 shadow-5'>
            <h1>Robot Card</h1>
            <img alt='robots' src= {`https://robohash.org/${id}?200x20`}/>
            <div>
                <h2>{name}</h2>
                <p>{email}</p>
            </div>
        </div>
    );
} 

export default Card;