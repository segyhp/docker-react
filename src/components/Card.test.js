import { shallow } from 'enzyme';
import React from 'react';
import Card from './Card';
import toJson, {createSerializer} from 'enzyme-to-json';


it('expect to render Card Component', () => {
    //what shallow do is create simple component
    //what mount do is create full component and need dom API to work or browser /headless browser
    //what render do is between shallow and mount, just for render but not fully functional
    // expect(shallow(<Card />).length).toEqual(1);
    //create snapshot
    expect(toJson(shallow(<Card />))).toMatchSnapshot();
});
