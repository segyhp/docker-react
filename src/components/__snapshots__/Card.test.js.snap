// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render Card Component 1`] = `
<div
  className="tc bg-light-green dib br3 pd3 ma2 grow bw2 shadow-5"
>
  <h1>
    Robot Card
  </h1>
  <img
    alt="robots"
    src="https://robohash.org/undefined?200x200"
  />
  <div>
    <h2 />
    <p />
  </div>
</div>
`;
